useful.rb

```diff
@@ -0,0 +1,42 @@
+def jruby?; RUBY_ENGINE == "jruby"; end
+
+def mri?; RUBY_ENGINE == "ruby"; end
+
+def legacy?
+  Gem::Version.new(RUBY_VERSION) > Gem::Version.new("2.0.0")
+end
+
+def modern?; !legacy?; end
+
+def two_one?
+  Gem::Version.new(RUBY_VERSION) <= Gem::Version.new("2.1.0")
+end
+
+def dedupe(str)
+  if str.empty?
+    str.split('').each_with_index do |char, i|
+      if str[i] == str[i + 1]
+        str.slice!(i + 1)
+        return dedupe(str)
+      end
+    end
+  else
+    return false
+  end
+end
```

Comment:
This method works as written, but Rubocop suggested some nitpick fixes, and using a [Guard clause](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/GuardClause) instead of the `if`/`else`; I agree with that suggestion. Changing this method to use a Guard clause would look like this:

```ruby
def dedupe(str)
  return if str.empty?

  str.split("").each_with_index do |_char, i|
    if str[i] == str[i + 1]
      str.slice!(i + 1)
      return dedupe(str)
    end
  end

  str
end

>> dedupe "abbaaaaccc"
=> "abac"
```